CMakeLists.txt

```diff
@@ -29,7 +29,7 @@ if(SPDLOG_USE_STD_FORMAT)
     set(CMAKE_CXX_STANDARD 20)
     set(CMAKE_CXX_STANDARD_REQUIRED ON)
 elseif(NOT CMAKE_CXX_STANDARD)
-    set(CMAKE_CXX_STANDARD 11)
+    set(CMAKE_CXX_STANDARD 14)
```

Comment:
This breaks backward compatibility. In amy case i dont think unique ptr should be involved. see the usage example in https://en.cppreference.com/w/cpp/thread/promise

Reply:
Sure. I was thinking that using a ptr would make it more flexible so that other kind of log message (e.g., terminate) could add a future dynamically. But let's see whether we would need it in the future.